site/src/main/java/uk/nhs/digital/ps/beans/Dataset.java

```diff
@@ -55,23 +55,23 @@ public String getGeographicCoverage() {
     }
 
     /*
-     * Migration: 20171208
-     * New field: publicationsystem:Files-v2
-     * Old field: publicationsystem:Files
+     * Migration: 20171213
+     * New field: publicationsystem:Files-v3
+     * Old field: publicationsystem:Files-v2
      *
-     * Temporary make this getter forward and backward compatible, meaning we can deploy this code without running
-     * content migration.
+     * Temporary supporting old and new versions until all the live content is migrated to the new way
      */
+    @HippoEssentialsGenerated(internalName = PropertyKeys.FILES_V3)
+    public List<Attachment> getFiles() {
+        assertPropertyPermitted(PropertyKeys.FILES_V3);
```

Comment:
We have a method `getChildBeansIfPermitted` in `Publication` which wraps calls to `assertPropertyPermitted` and `getChildBeansByName` in one - how about replicating it here?

To avoid code duplication, it could be added to our base class; as it relies on a call to `assertPropertyPermitted`, the latter could be added as a template, no-op method to the base class, overriden by extending classes.

Reply:
Good idea, will implement this.